torch/csrc/autograd/record_function_ops.cpp

```diff
@@ -84,12 +86,14 @@ c10::AliasAnalysisKind aliasAnalysisFromSchema() {
 
 jit::RegisterOperators reg_fut_ops({
     jit::Operator(
-        "profiler::_call_end_callbacks_on_jit_fut(Tensor x, Future(t) y) -> ()",
+        "profiler::_call_end_callbacks_on_jit_fut(Tensor x, Future(t) y) -> Future(t)",
         [](jit::Stack& stack) {
           // Pop inputs, which should be a future and a tensor
           auto fut = jit::pop(stack).toFuture();
           auto tensor = jit::pop(stack).toTensor();
-          _call_end_callbacks_on_fut(tensor, fut);
+          auto profiledFut = _call_end_callbacks_on_fut(tensor, fut);
+          // return future that completes when profiling callbacks have run.
+          jit::push(stack, profiledFut);
```

Comment:
IIUC, `push` takes rvalue reference. If so, shall we use `std::move()` on the `profilerFut`?